checks.d/kubernetes.py

```diff
@@ -235,25 +245,74 @@ def _update_container_metrics(self, instance, subcontainer, kube_labels):
         kube_labels = self.kubeutil.extract_kube_labels(pods_list, excluded_keys=excluded_labels)
 
         if not metrics:
             raise Exception('No metrics retrieved cmd=%s' % self.metrics_cmd)
 
+        # container metrics from Cadvisor
+        container_tags = {}
         for subcontainer in metrics:
+            c_id = subcontainer.get('id')
+            try:
+                tags = self._update_container_metrics(instance, subcontainer, kube_labels)
+                if c_id:
+                    container_tags[c_id] = tags
+                # also store tags for aliases
+                for alias in subcontainer.get('aliases', []):
+                    container_tags[alias] = tags
+            except Exception, e:
+                self.log.error("Unable to collect metrics for container: {0} ({1}".format(c_id, e))
+
+        # container metrics from kubernetes API: limits and requests
+        for pod in pods_list['items']:
             try:
-                self._update_container_metrics(instance, subcontainer, kube_labels)
-            except Exception as e:
-                self.log.error("Unable to collect metrics for container: {0} ({1}".format(
-                    subcontainer.get('name'), e))
+                containers = pod['spec']['containers']
+                name2id = {}
+                for cs in pod['status'].get('containerStatuses', []):
+                    c_id = cs.get('containerID', '').split('//')[-1]
+                    name = cs.get('name')
+                    if name:
+                        name2id[name] = c_id
+            except KeyError:
+                self.log.debug("Pod %s does not have containers specs, skipping...", pod['metadata'].get('name'))
+                continue
+
+            for container in containers:
+                c_name = container.get('name')
+                _tags = container_tags.get(name2id.get(c_name), [])
+
+                prog = re.compile(r'[-+]?\d+[\.]?\d*')
+
+                # limits
+                try:
+                    for limit, value_str in container['resources']['limits'].iteritems():
+                        values = [float(s) for s in prog.findall(value_str)]
+                        if len(values) != 1:
+                            self.log.warning("Error parsing limits value string: %s", value_str)
+                            continue
+                        self.publish_gauge(self, '{}.{}.limits'.format(NAMESPACE, limit), values[0], _tags)
+                except (KeyError, AttributeError) as e:
+                    self.log.debug("Unable to retrieve container limits for %s: %s", c_name, e)
+                    self.log.debug("Container object for {}: {}".format(c_name, container))
+
+                # requests
+                try:
+                    for request, value_str in container['resources']['requests'].iteritems():
+                        values = [float(s) for s in prog.findall(value_str)]
```

Comment:
FYI: the Kubernetes code dealing with quantities lives in 

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/resource/quantity.go
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/resource/suffix.go